### ISSUE TITLE
PostgreSQL support

### ISSUE THREAD

**Issue by adriancole** [openzipkin/zipkin-java#93]
@Koizumi85 asked on gitter about PostgreSQL support. spanstore-jdbc takes advantage of some SQL syntax which might not be supported in PostgreSQL. Before we start down this path, let's figure out some basics. For example, which version of PostgreSQL is needed, so that we can test the configuration desired.

**Comment by adriancole** [comment]
@Koizumi85 nag..

**Comment by eirslett** [comment]
Could it be supported as an external/optional module? I'm not sure it's a good idea to reuse the JDBC code, even if that's tempting.

**Comment by Koizumi85** [comment]
Actually i think with postgresql 9.5 the actual code could possibly work if there were a possibility to switch the database type in Jooq..
Background: psql9.5 supports "upsert" semantics which are heavily used here.
But postgresql 9.5 is fairly new and is not hosted in the repositories of Ubuntu LTS and other linux distributions and there are a lot of sysadmins who don't want to take the postgresql repositories for LTS reasons.

**Comment by eirslett** [comment]
9.1 or 9.3 are the safest choices, I think.

**Comment by Koizumi85** [comment]
9.1 and 9.3 should be almost identically unless the json column type of 9.3. but i don't think we would need this here, or?
So I think a 9.1 compatibility would be nice.

**Comment by eirslett** [comment]
Sounds good to me. The JSON query capabilites would be nice to have (for things like annotations), but it wasn't added in MySQL until 5.7.8, I think many linux distros are at MySQL 5.5 or so. And the existing SQL span store doesn't use JSON.

**Comment by adriancole** [comment]
This issue was moved to openzipkin/zipkin#1232
